### PR TITLE
[Relay][Op] Add operators full and full_like

### PR DESCRIPTION
**Pull request by slyubomirsky**
Pretty straightforward operators to add, but would appreciate any guidelines on style and on having negative test cases (really unsure about how to set those up).

### CODE REVIEW

File: src/relay/op/tensor/transform.cc [hunk 1]
```diff
@@ -76,6 +76,94 @@ RELAY_REGISTER_OP("expand_dims")
...
+.describe(R"code(Fill array with scalar value.
+
+)code" TVM_ADD_FILELINE)
+.set_num_inputs(1)
+.add_argument("fill_value", "double", "The value to fill.")
+.set_support_level(3)
+.add_type_rel("Full", FullRel);
+
+bool FullLikeRel(const Array<Type>& types,
+             int num_inputs,
```
Comment (MarisaKirisame) [inline src/relay/op/tensor/transform.cc]: indent

File: src/relay/op/tensor/transform.cc [hunk 2]
```diff
@@ -76,6 +76,94 @@ RELAY_REGISTER_OP("expand_dims")
...
+  CHECK_EQ(fill_value->shape.size(), 0)
+    << "The fill value should be a scalar but here it has dimension "
+    << fill_value->shape.size() << ".";
+
+  reporter->Assign(types[2], TensorTypeNode::make(data->shape, data->dtype));
+  return true;
+}
+
+Expr MakeFullLike(Expr data,
+              Expr fill_value) {
```
Comment (MarisaKirisame) [inline src/relay/op/tensor/transform.cc]: indent

**Comment by slyubomirsky** [comment]
Would you recommend making the fill value for `full` an attribute, as in `pad`? Don't know if in practice you'd want to set that programmatically

**Comment by tqchen** [comment]
Yah, I think having value as attribute might makes sense here

**Comment by MarisaKirisame** [comment]
Hi @slyubomirsky @tqchen , I was trying to implement zeros and ones, but it has code exactly the same as full. Is there anyway steven can get his record of contribution, and I am not blocked on it? I can copy the code to my branch but then it will be unfair to him.

**Comment by tqchen** [comment]
Let us trying to just work to merge this PR in. @slyubomirsky can you rebase?

**Comment by slyubomirsky** [comment]
Will rebase

**Comment by slyubomirsky** [comment]
Also realized that there is a problem with making the `fill_value` an attribute, as that now would mean that there is no data type to copy if the `dtype` attribute is left unspecified. I've left the code as it was but we can easily change it later

**Pull request merged**